src/overlays/actors/ovl_Bg_Haka_Gate/z_bg_haka_gate.c

```diff
@@ -46,33 +68,306 @@ const ActorInit Bg_Haka_Gate_InitVars = {
+void BgHakaGate_GateOpen(BgHakaGate* this, GlobalContext* globalCtx) {
+    if (Math_StepToF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y + 80.0f, 1.0f)) {
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_METALDOOR_STOP);
+        this->dyna.actor.flags &= ~0x10;
+        this->actionFunc = BgHakaGate_DoNothing;
+    } else {
+        func_8002F974(&this->dyna.actor, NA_SE_EV_METALDOOR_SLIDE - SFX_FLAG);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C158.s")
+void BgHakaGate_SkullOfTruth(BgHakaGate* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->switchFlag) && Math_StepToS(&this->vFlameScale, 350, 20)) {
+        this->actionFunc = BgHakaGate_DoNothing;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C270.s")
+void BgHakaGate_FalseSkull(BgHakaGate* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        Math_StepToS(&this->vFlameScale, 350, 20);
+    }
+    if (globalCtx->actorCtx.unk_03) {
+        this->dyna.actor.flags |= 0x80;
+    } else {
+        this->dyna.actor.flags &= ~0x80;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C454.s")
+void BgHakaGate_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgHakaGate* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C5D0.s")
+    this->actionFunc(this, globalCtx);
+    if (this->dyna.actor.params == BGHAKAGATE_SKULL) {
+        this->vScrollTimer++;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C65C.s")
+void BgHakaGate_DrawFlame(BgHakaGate* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    f32 scale;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C6AC.s")
+    if (this->vFlameScale > 0) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_haka_gate.c", 716);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C73C.s")
+        if (1) {}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/func_8087C794.s")
+        func_80093D84(globalCtx->state.gfxCtx);
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0,
+                                    (this->vScrollTimer * -20) & 0x1FF, 0x20, 0x80));
+        gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 255, 255, 0, 255);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 0, 0, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Gate/BgHakaGate_Update.s")
+        Matrix_Translate(thisx->posRot.pos.x, thisx->posRot.pos.y + 15.0f, thisx->posRot.pos.z, MTXMODE_NEW);
+        Matrix_RotateY(Camera_GetCamDirYaw(ACTIVE_CAM) * (M_PI / 0x8000), 1);
```

Comment:
There are a few mtx enums missing in these 2 draw functions.
```suggestion
        Matrix_RotateY(Camera_GetCamDirYaw(ACTIVE_CAM) * (M_PI / 0x8000), MTXMODE_APPLY);
```